### PR TITLE
Switch to Semantic Convention 1.27.0 Adding New Outbox Activities

### PR DESCRIPTION
**Pull request by oslaksam**
- Added Outbox Producer Activity
- Added Entry Created Activity
- Added Unit Tests For These Activities
- Changed How Outbox Enqueue Activity Starts
- Changed Activity Name Semantic Convention to 1.27.0

### CODE REVIEW

**Review by toizo — COMMENTED** [review]
Looks like a good work. Just minor comments. 

However, I myself would prefer a two separate pull requests, where one adds an Outbox producer instrumentation and the other enhances how the enqueuing activities are registered - so there are fewer moving pieces for a PR

File: src/Dafda/Diagnostics/OpenTelemetryMessagingOperation.cs [hunk 1]
```diff
@@ -16,5 +21,11 @@ internal static class Consumer
         /// One or more messages are requested by a consumer. This operation refers to pull-based scenarios, where consumers explicitly call methods of messaging SDKs to receive messages.
         /// </summary>
         public const string Receive = "receive";
+
+
+        /// <summary>
+        /// One or more messages are passed to a consumer. This operation refers to push-based scenarios, where consumer register callbacks which get called by messaging SDKs.
+        /// </summary>
+        public const string Deliver = "deliver";
```
[the remark this answers is not in the document]
Comment (oslaksam) [reply src/Dafda/Diagnostics/OpenTelemetryMessagingOperation.cs:29]: No copilot generated it from the specification of operation names:
https://github.com/open-telemetry/semantic-conventions/blob/v1.24.0/docs/messaging/messaging-spans.md

File: src/Dafda/Outbox/OutboxQueue.cs [hunk 2]
```diff
@@ -62,7 +65,14 @@ public async Task<IOutboxNotifier> Enqueue(IEnumerable<object> messages)
         /// </remarks>
         public async Task<IOutboxNotifier> Enqueue(IEnumerable<object> messages, Metadata headers)
```
[the remark this answers is not in the document]
Comment (oslaksam) [reply src/Dafda/Outbox/OutboxQueue.cs:66]: Good suggestion I will change this.

File: src/Dafda/Diagnostics/OpenTelemetryMessagingOperation.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (oslaksam) [reply src/Dafda/Diagnostics/OpenTelemetryMessagingOperation.cs]: Removed

**Review by toizo — APPROVED** [review]

**Pull request merged**